metadata/ota_metadata/metadata_gen.py

```diff
@@ -147,12 +193,27 @@ def gen_metadata(
             stat = os.stat(os.path.join(target_dir, d))
             nlink = stat.st_nlink
             inode = stat.st_ino if nlink > 1 else ""
+            sha256hash = _file_sha256(os.path.join(target_dir, d))
+
+            # if compression is enabled, try to compress the file here
+            compress_alg = ""
+            if compression_enabled:
+                if zstd_compress_file(
+                    cctx,
+                    os.path.join(target_dir, d),
+                    os.path.join(compressed_dir, sha256hash),
+                    cmpr_ratio=cmpr_ratio,
+                    filesize_threshold=filesize_threshold,
+                ):
+                    compress_alg = f",{ZSTD_COMPRESSION_ALG}"
```

Comment:
Adding optimization.
```suggestion
            if compression_enabled:
                src_file = os.path.join(target_dir, d)
                dst_file = os.path.join(compressed_dir, sha256hash)
                if os.path.exists(dst_file) or zstd_compress_file(
                      cctx,
                      src_file,
                      dst_file,
                      os.path.join(compressed_dir, sha256hash),
                      cmpr_ratio=cmpr_ratio,
                      filesize_threshold=filesize_threshold,
                ):
                    compress_alg = ZSTD_COMPRESSION_ALG
                else:
                    compress_alg = ""
```

Reply:
@Bodong-Yang 
See above optimization.

Reply:
Updated in [b2dc29a](https://github.com/tier4/ota-metadata/pull/12/commits/b2dc29a538de74ab3e55d21d4d6d1bbba8370c0e).